stream_alert/alert_processor/outputs.py

```diff
@@ -803,3 +805,272 @@ def dispatch(self, **kwargs):
+        params = {'jql': jql,
+                  'maxResults': max_results,
+                  'validateQuery': validate_query,
+                  'fields': fields}
+        issues = []
+
+        resp = self._get_request(search_url, params=params, headers=headers, verify=False)
+
+        try:
+            issues = resp.json()['issues']
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while searching Jira [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+
+        return issues
+
+    def _create_comment(self, issue_id, comment):
+        """Add a comment to an existing issue
+
+        Args:
+            issue_id (str): The existing issue ID or key
+            comment (str): The body of the comment
+
+        Returns:
+            int: ID of the created comment or False if unsuccessful
+        """
+
+        comment_url = os.path.join(self.base_url, self.COMMENT.format(issue_id))
+        comment_body = {'body' : comment}
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
+        resp = self._post_request(comment_url, data=comment_body, headers=headers, verify=False)
+        jira_id = False
+
+        try:
+            jira_id = resp.json()['id']
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while creating Jira comment [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+        return jira_id
+
+
+    def _get_comments(self, issue_id):
+        """Get all comments for an existing Jira issue
+
+        Args:
+            issue_id (str): The existing issue ID or key
+
+        Returns:
+            list: List of comments associated with a Jira issue
+        """
+        comment_url = os.path.join(self.base_url, self.COMMENT.format(issue_id))
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
+        resp = self._get_request(comment_url, headers=headers, verify=False)
+        comments = []
+
+        try:
+            comments = resp.json()['comments']
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while getting Jira comments [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+        return comments
+
+    def _get_existing_issue(self, issue_summary, project_key):
+        """Find an existing Jira issue based on the issue summary
+
+        Args:
+            issue_summary (str): The Jira issue summary
+            project_key (str): The Jira project to search
+
+        Returns:
+            int: ID of the found issue or False if existing issue does not exist
+        """
+        jql = 'summary ~ "{}" and project="{}"'.format(issue_summary, project_key)
+        resp = self._search_jira(jql, fields=['id', 'summary'], max_results=1)
+        jira_id = False
+
+        try:
+            jira_id = resp[0]['id']
```

Comment:
Could this ever result in a `KeyError` as well (with `'id'`)? Genuinely curious (I don't have much jira experience)

Reply:
If a list of dicts is passed back, it would be fairly likely for the 'id' to be present, but this should definitely be modified to "except (IndexError, KeyError)" just in case.  Nice catch! (hehe)